plugins/coverimage.koplugin/main.lua

```diff
@@ -0,0 +1,176 @@
+local Device = require("device")
+
+if not Device.isAndroid() and not Device.isEmulator() then
+    return { disabled = true }
+end
+
+local DocSettings = require("docsettings")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local CoverImage = WidgetContainer:new{
+    name = 'coverimage',
+    is_doc_only = true,
+    settings_id = 0,
+}
+
+function CoverImage:_enabled()
+    return self.enabled
+end
+
+function CoverImage:_restore()
+    return self.enabled and self.restore
+end
+
+function CoverImage:init()
+    self.cover_image_path = G_reader_settings:readSetting("cover_image_path") or "Cover.png"
+    self.enabled = G_reader_settings:isTrue("cover_image_enabled")
+    self.restore = G_reader_settings:isTrue("cover_image_restore") and self.enabled
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function CoverImage:cleanUpImage()
+    local lfs = require("libs/libkoreader-lfs")
+    -- Delete image the backup (if it exits) will contain a user defined sleep image
+    if lfs.attributes(self.cover_image_path) then
+        os.remove(self.cover_image_path)
+    end
```

Comment:
As this is always called by onCloseDocument() (if restore=true), if there is no coverimage, this might delete the default image (which would not have been renamed to .bak) ?